### PR TITLE
Added group `name`

### PR DESCRIPTION
**Pull request by RichardLitt**
I was unable to get an argument working with resolve. That should work.

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -1661,9 +1661,261 @@ root will not be listable, as it is virtual. Access known paths directly.
 
 # Group name
 
-## publish
+IPFS namespace (IPNS) tool
```
Comment (dignifiedquire) [inline apiary.apib]: Link to ipns spec/description/docs would be good

File: apiary.apib [hunk 2]
```diff
@@ -1661,9 +1661,261 @@ root will not be listable, as it is virtual. Access known paths directly.
...
 
-## resolve
+IPNS is a PKI namespace, where names are the hashes of public keys, and
+the private key enables publishing new (signed) values. In publish, the
+default value of <name> is your own identity public key.
+
+This command cannot be called directly.
+
+## publish [POST /name/publish{?arg}{&resolve,lifetime,ttl}]
+Publish an object to IPNS
```
Comment (dignifiedquire) [inline apiary.apib]: Missing `.`

File: apiary.apib [hunk 3]
```diff
@@ -1661,9 +1661,261 @@ root will not be listable, as it is virtual. Access known paths directly.
...
+
++ Parameters
+    + arg (string, required) - IPFS path of the object to be published
+    + resolve (boolean, optional) - Resolve given path before publishing
+        + Default: `true`
+    + lifetime (string, optional) - Time duration that the record will be valid for. 
+        This accepts durations such as '300s', '1.5h' or '2h45m'. Valid time units are
+        'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'. Default: '24h'.
+    + ttl (string, optional) - Time duration this record should be cached for. Caution: experimental.
```
Comment (dignifiedquire) [inline apiary.apib]: What format does this have? The same as `lifetime`, if so would be good to have a custom type of `time` for these.

File: apiary.apib [hunk 4]
```diff
@@ -1661,9 +1661,261 @@ root will not be listable, as it is virtual. Access known paths directly.
...
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (object)
+        + Path (MultiAddr)
+
+    + Body
+
+        ```
+        {"Path":"/ipfs/QmW2WQi7j6c7UgJTarActp7tDNikE4B2qXtFCfLPdsgaTQ"}
```
Comment (dignifiedquire) [inline apiary.apib]: Let's make this multiline like other examples:

```
{
  "Path": "/ipfs/QmW2WQi7j6c7UgJTarActp7tDNikE4B2qXtFCfLPdsgaTQ"
}
```

File: apiary.apib (hunk 1 above)
Comment (RichardLitt) [inline apiary.apib]: Not sure about that. I'd rather not add links which aren't part of the CLI mans.

**Comment by RichardLitt** [comment]
@dignifiedquire added a `time` type. Thoughts?

**Comment by RichardLitt** [comment]
Note that `resolve` defaults are not yet in the IPFS API. Holding off on PRing until fixed https://github.com/ipfs/go-ipfs/pull/2424

File: apiary.apib [hunk 5]
```diff
@@ -4428,6 +4660,20 @@ The IPFS error object
         + `0`
         + `1`
 
+## Time (string)
+
+The time format used in some commands. Note that this is not technically an enumerated type, as multiple
+time formats can be joined. For example, '3h30m'.
+
++ (string) - The time format, composed of a number followed by an enumerated list with the following members.
+    + Members
```
Comment (dignifiedquire) [inline apiary.apib]: If it's not an enum `Members` is wrong afaik.

Comment (RichardLitt) [inline apiary.apib]: Not sure how else to do this. Just a text description?

Comment (dignifiedquire) [inline apiary.apib]: yeah

Comment (RichardLitt) [inline apiary.apib]: Updated.

**Comment by dignifiedquire** [comment]
LGTM

**Pull request merged**